### PR TITLE
J3RN's solutions

### PR DESCRIPTION
**Pull request by J3RN**
This took me far longer than it should have.
### Things to Think About
- I would first test the ability to split strings effectively, followed by counting words correctly.
- The edge cases, for me, were apostrophes. Not technically a word character, so words containing apostrophes were split around the apostrophe.
- My solution is, in fact, object-oriented, but arguably needlessly so.

### CODE REVIEW

**Comment by jaybobo** [comment]
Nice. :+1:

File: counter.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    contents = File.read(@filename)
+    word_hash = split_text(contents)
+    print_descending(word_hash)
+  end
+
+  # Splits the text into a hash of words and counts
+  def split_text(text)
+    all_words = text.scan(/((?:\w+|\')+)/).flatten.map { |x| x.downcase }
+
+    mapping = Hash.new
```
Comment (jaybobo) [inline counter.rb]: Anytime I see a pattern where we initialize a thing, do something to it, and return it...I always ask myself could I have used `.tap` or `.reduce`.

File: counter.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  # Do everything
+  def run
+    contents = File.read(@filename)
+    word_hash = split_text(contents)
+    print_descending(word_hash)
+  end
+
+  # Splits the text into a hash of words and counts
```
Comment (jaybobo) [inline counter.rb]: using `private` here is probably a good idea. it signals to other devs that the methods below are "dependencies."  (i.e. internal methods that we wouldn't test.)

File: counter.rb (hunk 1 above)
Comment (J3RN) [inline counter.rb]: Ideally, I would do a `map`, followed with a `to_h`, but evidently Ruby 2.0.0 does not have `Array#to_h`.

I can try to do a similar thing with `.tap` or `.reduce`
